spec/controllers/subjects_controller_spec.rb

```diff
@@ -0,0 +1,76 @@
+require 'rails_helper'
+
+include Warden::Test::Helpers
+Warden.test_mode!
+
+RSpec.describe SubjectsController, type: :controller do
+  render_views
+
+  describe 'GET #new' do
+    before do
+      allow(controller).to receive(:user_signed_in?).and_return(false)
+      allow(controller).to receive(:current_user).and_return(false)
+    end
+
+    context 'when user is unauthenticated' do
+      it 'redirects to login' do
+        get :new
+        expect(response).to redirect_to unauthenticated_root_url
+      end
+    end
+
+    context 'when user is authenticated' do
+      it 'assigns @subject to new' do
+        allow(controller).to receive(:user_signed_in?).and_return(true)
+        get :new
+        expect(assigns(:subject)).to be_a_new(Subject)
+      end
+
+      it "renders the subject template" do
+        allow(controller).to receive(:user_signed_in?).and_return(true)
+        get :new
+        expect(response).to render_template('subjects/new')
+      end
+    end
+
+    let(:user) { FactoryBot.create(:user) }
```

Comment:
keep all `let` and `before` statements in one place so it will be easier to track *all* context changes

Reply:
Fixed, if I correctly understood you, also fixed the FactoryBot to a shorter version (which you pointed out before)